config/services.php

```diff
@@ -38,9 +40,25 @@
         ->tag('console.command');
 
     $services
-        ->set('mongodb.prototype.client', Client::class)
+        ->set('mongodb.abstract.client', Client::class)
         ->arg('$uri', abstract_arg('Should be defined by pass'))
         ->arg('$uriOptions', abstract_arg('Should be defined by pass'))
         ->arg('$driverOptions', abstract_arg('Should be defined by pass'))
-        ->tag('mongodb.client');
+        ->abstract();
```

Comment:
@OskarStark I think using an abstract service is more in the Symfony than having a prototype definition that needs to be removed.

In any case, I think it's a good pattern, allowing users to modify the basic definition to change the class, for example.

Reply:
I agree 👍

Reply:
Yes, this is a lot nicer than my approach of using a regular service that can cause issues when it's not removed.